datalad/support/archives.py

```diff
@@ -10,55 +10,166 @@
+    ----------
+    archive: str
+    dir_: str
+    leading_directories: {'strip', None}
+    """
+    if not exists(dir_):
+        lgr.debug("Creating directory %s to extract archive into" % dir_)
+        os.makedirs(dir_)
 
-def decompress_file(file, dir, directories='strip'):
-    fullcmd = None
-    for ptr, cmd in DECOMPRESSORS.iteritems():
-        if re.search(ptr, file):
-            fullcmd = cmd % locals()
-            break
-#    if fullcmd is not None:
-#        lgr.debug("Extracting file: %s" % fullcmd)
-#        status, output = getstatusoutput(fullcmd)  # getstatusoutput is deprecated. Use cmd.Runner.run() instead.
-#        if status:
-#            lgr.debug("Failed to extract: status %d output %s" % (status, output))
-#    else:
-    #lgr.debug("Have no clue how to extract %s -- using patool" % file)
-    verbosity = -1                        # silent by default
-    ef_level = lgr.getEffectiveLevel() 
-    if ef_level and lgr.getEffectiveLevel() <= logging.DEBUG:
-        verbosity = 1
-    #elif lgr.getEffectiveLevel() <= logging.INFO:
-    #    verbosity = 0
-    patoolib.extract_archive(file, outdir=dir, verbosity=verbosity)
-    if directories == 'strip':
-        _, dirs, files = os.walk(dir).next()
+    from ..tests.utils import swallow_outputs
+    with swallow_outputs() as cmo:
+        patoolib.util.check_existing_filename(archive)
+        patoolib.util.check_existing_filename(dir_, onlyfiles=False)
+        # Call protected one to avoid the checks on existence on unixified path
+        patoolib._extract_archive(unixify_path(archive),
+                                  outdir=unixify_path(dir_),
+                                  verbosity=100)
+        if cmo.out:
+            lgr.debug("patool gave stdout:\n%s" % cmo.out)
+        if cmo.err:
+            lgr.debug("patool gave stderr:\n%s" % cmo.err)
+
+    if leading_directories == 'strip':
+        _, dirs, files = os.walk(dir_).next()
         if not len(files) and len(dirs) == 1:
             # move all the content under dirs[0] up 1 level
-            subdir, subdirs_, files_ = os.walk(join(dir, dirs[0])).next()
+            widow_dir = opj(dir_, dirs[0])
+            lgr.debug("Moving content within %s upstairs" % widow_dir)
+            subdir, subdirs_, files_ = os.walk(opj(dir_, dirs[0])).next()
             for f in subdirs_ + files_:
-                os.rename(join(subdir, f), join(dir, f))
+                os.rename(opj(subdir, f), opj(dir_, f))
+            os.rmdir(widow_dir)
+    elif leading_directories is None:
+        pass   # really do nothing
     else:
-        raise NotImplementedError("Not supported %s" % directories)
-    # TODO: (re)move containing directory
+        raise NotImplementedError("Not supported %s" % leading_directories)
+
+
+def compress_files(files, archive, path=None, overwrite=True):
+    """Compress `files` into an `archive` file
+
+    Parameters
+    ----------
+    files : list of str
+    archive : str
```

Comment:
'path' and 'overwrite' missing. From just looking at the signature I'm not sure what these are for.